app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,34 @@
+class PasswordResetsController < ApplicationController
+  def new
+  end
+  def create
+    user = User.find_by_email(params[:email])
+    if user
+      user.send_password_reset 
+      redirect_to root_url, notice: "Email sent with password reset instructions."
+    else
+      redirect_to new_password_reset_path, alert: "We couldn't find this user."
+    end
+  end
+
+  def edit
+    #id here is password reset token
```

Comment:
For lines 16 and 20, consider modifying `config/routes.rb` so the param name is the more descriptivate `:token` instead of `:id`: http://edgeguides.rubyonrails.org/routing.html#overriding-named-route-parameters